core/handler/device/store.go

```diff
@@ -23,11 +24,13 @@ type Store interface {
 	DownlinkQueue(appID, devID string) (DownlinkQueue, error)
 	Set(new *Device, properties ...string) (err error)
 	Delete(appID, devID string) error
+	AttrWhitelist() map[string]bool
 }
 
 const defaultRedisPrefix = "handler"
 const redisDevicePrefix = "device"
 const redisDownlinkQueuePrefix = "downlink"
+const redisUpStreamMetadataKeyWhitelist = ":attr-whitelist"
```

Comment:
1. Is there is one already or built one ?
2. 128 or 256 ?
3. That's the white list, name can be anything as long as it state that it's always allow this attributes. `common attributes` / `ttn attributes`
4. The limitation is implemented, have to take it from the config. Sort the pushed attributes, you mean sort the attributes received by the client and then push them to the db ?